src/validators/db-lookup.validator.ts

```diff
@@ -17,55 +17,45 @@ export class DbLookupValidator implements ValidatorConstraintInterface {
       ignoreEmpty = true,
       type,
       findOption,
+      validateNumeric = false,
     }: DbLookupOptions<BaseEntity> = args.constraints[0];
 
-    // Handle empty or null values
     if (value || !ignoreEmpty) {
-      // Additional check for non-numeric input (assuming IDs are numeric)
-      if (isNaN(value)) {
-        throw new BadRequestException(`Invalid input: "${value}" is not a valid numeric identifier.`);
+      if (validateNumeric && isNaN(value)) {
+        throw new BadRequestException(
+          `Invalid input: "${value}" must be a valid numeric value for this field.`
+        );
       }
 
       const options =
         findOption === 'primary'
           ? {
-              // Find by the entity's primary key
```

Comment:
Could just be me, but I think this comment is helpful.

Reply:
``` markdown
Agreed. Updated the validation logic on the revised version.